tools/devops/automation/scripts/bash/compare.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+#
+# How to run API comparison locally:
+#
+# 1. To compare a specific commit to the previous one, just run this script (./compare.sh).
+# 2. Replicating what happens for a specific PR is a bit more complicated:
+#    a. Fetch the source code for that PR, using "git fetch --no-tags --progress https://github.com/xamarin/xamarin-macios +refs/pull/<PR>/*:refs/remotes/origin/pr/<PR>/*",
+#       so for example this for PR #4614:
+#
+#            git fetch --no-tags --progress https://github.com/xamarin/xamarin-macios +refs/pull/4614/*:refs/remotes/origin/pr/4614/*
+#
+#    b. Checkout the commit you want to compare (typically the branch you created the PR from if it's your own PR).
+#       i) This isn't 100% accurate, since GitHub will build a merged version
+#          of your branch + the target branch. If you want to test _exactly_
+#          the same code as GitHub, checkout the 'origin/pr/<merge>/merge'
+#          branch the previous git fetch command created.
+#
+#    c. Execute 'ghprbPullId=4614 ./compare.sh'
+#
+
+WORKSPACE="$BUILD_ARTIFACTSTAGINGDIRECTORY"
+API_COMPARISON="$WORKSPACE/apicomparison"
+BUILD_URL="$SYSTEM_TEAMFOUNDATIONCOLLECTIONURI$SYSTEM_TEAMPROJECT/_build/index?buildId=$BUILD_BUILDID&view=ms.vss-test-web.test-result-details"
+URL_PREFIX="$VSDROPSPREFIX/$BUILD_BUILDNUMBER/$BUILD_BUILDID/apigeneratordiff/;/"
+API_URL="$URL_PREFIX/api-diff.html"
+GENERATOR_URL="$URL_PREFIX/generator-diff/index.html"
+
+# env var should have been defined by the CI
+if test -z "$XAM_TOP"; then
+    echo "Variable XAM_TOP is missing."
+    exit 1
+fi
```

Comment:
This contradicts the documentation at the top: `To compare a specific commit to the previous one, just run this script (./compare.sh).`

In fact, the changes in this script looks like it won't be easy to run locally. Did you test running it locally to just compare with the previous commit?

Reply:
Need to fix that